components/MapViewScreen/index.tsx

```diff
@@ -52,19 +52,43 @@ export default function MapViewScreen({
     location: [],
   });
 
-  const handleFilterChange = (filter: FilterType) => {
-    console.log(filter);
+  const [filteredProjectsWithSearch, setFilteredProjectsWithSearch] =
+    useState<Project[]>(projects);
+
+  // show projects based on selected filters
+  const handleFilterButtonClick = () => {
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    const { status, technology, projectSize, location } = selectedFilters;
+    // add all filtering logic here
+    const filteredProjects = projects?.filter(project =>
+      technology.includes(project.renewable_energy_technology),
+    );
+    setFilteredProjects(filteredProjects);
+  };
+
+  // clear filters
+  const clearFilters = () => {
+    setSelectedFilters({
+      status: [],
+      technology: [],
+      projectSize: { min: 0, max: 0 },
+      location: [],
+    });
+    setFilteredProjects(projects);
   };
 
+  // search within all projects
   useEffect(() => {
-    let filtered: Project[] = [];
-    filtered =
+    const searchedProjects: Project[] =
       projects?.filter(project =>
         project.project_name.toLowerCase().includes(searchTerm.toLowerCase()),
-      ) ?? null;
+      ) ?? [];
+    setFilteredProjectsWithSearch(searchedProjects);
+  }, [projects, searchTerm]);
 
-    setFilteredProjects(filtered);
-  }, [projects, searchTerm, setFilteredProjects]);
+  const handleFilterChange = (filter: FilterType) => {
+    console.log(filter);
```

Comment:
Please delete all console.log statements. As I mentioned before, we don't want them in our deployed app.